library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemAdapter.java

```diff
@@ -82,8 +82,9 @@ public void replaceEntities(List<MessageListItem> newEntities) {
                 new MessageListItemDiffCallback(messageListItemList, newEntities), true);
 
         // only update those rows that change...
-        result.dispatchUpdatesTo(this);
+
         messageListItemList = newEntities;
+        result.dispatchUpdatesTo(this);
     }
 
```

Comment:
isn't it better to first dispatchUpdates and after that swap the variable?

Reply:
Yep, that is the normal, but not important.